### PR TITLE
Refactor ClearStateAndRedirect

### LINKED ISSUE #47: Refactor session expired section
Remove duplications, deal with the "redirect" function

### PR DESCRIPTION
**Pull request by Danielsio**
## Changes Summery: 
1. Refactor ClearStateAndRedirect to be implemented in Context.
2. Add navigation to /generate-calendar after successfully login.
3. fix import of skeleton component from `@mui/lab` to `@mui/material`

## Relevent Issues:
close #47

### CODE REVIEW

File: src/context/UserContext.jsx [hunk 1]
```diff
@@ -115,12 +123,14 @@ const UserProvider = ({children}) => {
                 if (response.status === 200 && response.data.details === LOGIN) {
                     console.log("inside if case of 'Login'");
                     setIsCompletedFirstSetup(true);
+                    navigate("/generate-calendar")
```
Comment (oshriFadlon1) [inline src/context/UserContext.jsx:126]: need to comment the prints

File: src/context/UserContext.jsx [hunk 2]
```diff
@@ -115,12 +123,14 @@ const UserProvider = ({children}) => {
...
                 } else if (
                     response.status === 201 &&
                     response.data.details === REGISTER
                 ) {
                     console.log("inside if case of 'Register'");
```
Comment (oshriFadlon1) [inline src/context/UserContext.jsx:131]: same here

File: src/context/UserContext.jsx (hunk 1 above)
Comment (Danielsio) [reply src/context/UserContext.jsx:125]: removed

File: src/context/UserContext.jsx (hunk 2 above)
Comment (Danielsio) [reply src/context/UserContext.jsx]: sure no problem

**Review by itayf9 — APPROVED** [review]
All good

**Pull request merged**